ocs_ci/ocs/amq.py

```diff
@@ -0,0 +1,216 @@
+            raise cf
+        time.sleep(5)
+        if self.is_amq_pod_running(pod_pattern="zookeeper"):
+            return self.kafka_persistent
+        else:
+            raise ResourceWrongStatusException("my-cluster-zookeeper Pod is not getting to running state")
+
+    def setup_amq_kafka_connect(self):
+        """
+        The function is to setup amq-kafka-connect, the yaml file is pulling from github
+        it will make kind: KafkaConnect and will make sure the status is running
+        :return: kafka_connect
+        """
+        try:
+            kafka_connect = templating.load_yaml(os.path.join(self.dir, self.amq_kafka_connect_yaml))
+            self.kafka_connect = OCS(**kafka_connect)
+            self.kafka_connect.create()
+        except(CommandFailed, CalledProcessError) as cf:
+            log.error('Failed during setup of AMQ KafkaConnect')
+            raise cf
+
+        if self.is_amq_pod_running(pod_pattern="my-connect-cluster-connect"):
+            return self.kafka_connect
+        else:
+            raise ResourceWrongStatusException("my-connect-cluster-connect pod is not getting to running state")
+
+    def setup_amq_kafka_bridge(self):
+        """
+        Function to setup amq-kafka, the file file is pulling from github
+        it will make kind: KafkaBridge and will make sure the pod status is running
+        :return: kafka_bridge
+        """
+        try:
+            kafka_bridge = templating.load_yaml(os.path.join(self.dir, self.amq_kafka_bridge_yaml))
+            self.kafka_bridge = OCS(**kafka_bridge)
+            self.kafka_bridge.create()
+        except(CommandFailed, CalledProcessError) as cf:
+            log.error('Failed during setup of AMQ KafkaConnect')
+            raise cf
+        ## Making sure the kafka_bridge is running
+        if self.is_amq_pod_running(pod_pattern="my-bridge-bridge"):
+            return self.kafka_bridge
+        else:
+            raise ResourceWrongStatusException("kafka_bridge_pod pod is not getting to running state")
+
+    def setup_amq(self):
+        """
+        Setup AMQ from local folder,
+        function will call all necessary sub functions to make sure amq installation is complete
+        """
+        self.setup_amq_cluster_operator()
+        self.setup_amq_kafka_persistent()
+        self.setup_amq_kafka_connect()
+        self.setup_amq_kafka_bridge()
+        self.amq_is_setup = True
+        return self
+
+    def cleanup(self):
+        """
+        Clean up function,
+        will start to delete from amq cluster operator
+        then amq-connector, persistent, bridge, at the end it will delete the created namespace
+        """
+        if self.amq_is_setup:
+            self.kafka_persistent.delete()
+            self.kafka_connect.delete()
+            self.kafka_bridge.delete()
+            run_cmd(f'oc delete -f {self.amq_dir}', shell=True, check=True, cwd=self.dir)
```

Comment:
Use exec_oc_cmd() in ocs-ci/ocs_ci/ocs/ocp.py OCP class for executing all oc commands.